### ISSUE TITLE
[BUG] tf.layers.dropout not working

### ISSUE THREAD

**Issue by ghostplant** [tensorflow/tensorflow#30301]
To reproduce the issue:
```py
sess = tf.Session()
x = tf.ones([4, 4])
y = tf.layers.dropout(x, 0.5)
sess.run(y)
```

This version works well:
```py
sess = tf.Session()
x = tf.ones([4, 4])
y = tf.nn.dropout(x, 0.5)
sess.run(y)
```

Tensorflow version: 1.13.3
OS:  Ubuntu 18.04

**Comment by Leslie-Fang** [comment]
I think this API tf.layers.dropout is already deprecated

**Comment by Leslie-Fang** [comment]
I think tf.layers.dropout would use keras.layers.Dropout
And finally call this "lambda: array_ops.identity(inputs)" to return the exactly same tensor

**Comment by ghostplant** [comment]
It would be more reasonable if it is wrapped to `tf.nn.dropout`

**Comment by gadagashwini** [comment]
@ghostplant I tried executing the code on Colab with Tensorflow 1.13.1
```
import tensorflow as tf
sess = tf.Session()
x = tf.ones([4, 4])
y = tf.layers.dropout(x, 0.5)
sess.run(y)

```
Is this the expected output
array([[1., 1., 1., 1.],
       [1., 1., 1., 1.],
       [1., 1., 1., 1.],
       [1., 1., 1., 1.]], dtype=float32)
Thanks!

**Comment by ghostplant** [comment]
@gadagashwini No, if you use `tf.nn.dropout` instead, you should get the expected output: somes are zero and others are two.

**Comment by gadagashwini** [comment]
@ghostplant You are correct. With `tf.nn.dropout` i am able to get expected result. Thanks!

**Comment by yongtang** [comment]
The following will work:
```
import tensorflow as tf
sess = tf.Session()
x = tf.ones([4, 4])
y = tf.layers.dropout(x, 0.5, training=True)
sess.run(y)
```

Note in the doctoring of `tf.layers.dropout`:
```
    training: Either a Python boolean, or a TensorFlow boolean scalar tensor
      (e.g. a placeholder). Whether to return the output in training mode
      (apply dropout) or in inference mode (return the input untouched).
```

**Comment by ghostplant** [comment]
@yongtang Thank you. It is a little weird that training is set to False by default, because it is just not what this operator describes to be, so I think it would be more reasonable to change the default value. I think many developers don't even notice this case when they just select to use a dropout operator.
